public/app/features/panel/panellinks/cellDisplayValuesProxy.ts

```diff
@@ -0,0 +1,35 @@
+import { DisplayValue, DataFrame, formattedValueToString, getDisplayProcessor } from '@grafana/data';
+import { config } from '@grafana/runtime';
+import toNumber from 'lodash/toNumber';
+
+export function getCellDisplayValuesProxy(frame: DataFrame, rowIndex: number): Record<string, DisplayValue> {
+  return new Proxy({} as Record<string, DisplayValue>, {
```

Comment:
Really like the proxy approach!

Reply:
much much better than creating a variable for every option like we do here:  https://github.com/grafana/grafana/blob/master/packages/grafana-data/src/field/fieldDisplay.ts#L139

Reply:
But we still need to enumerate all the options and actually create them in https://github.com/grafana/grafana/pull/21478/files#diff-4468324e53509e365e2af108e45884c5R117 right? So what this gives us is access that is not mentioned in the autocomplete options (like use title or index of the field instead of name when manually writing the variable). I assume that is by design?

Reply:
That enumeration only happens in the editor, it does not happen every time we render the links.

I did not add title and index examples to the suggestions -- adding it for everything would be verbose -- maybe adding suggestions that look imply you fill in the name